### DISCUSSION TITLE
gunicorn worker getting timeout

### DISCUSSION THREAD

**Discussion by mainulquraishi** [miguelgrinberg/Flask-SocketIO#1614] [Q&A]
Hello, 
I am sending a data stream. Sometimes the client getting disconnected but reconnected again. 

I looked into `gunicorn` log and it says
`worker timeout`
then a new worker starts. I don't know, **why in the middle of the WebSocket connection, the worker would timeout?** 

I changed the `gunicorn` `--timeout` parameter to 90, I don't know, it will solve the problem, and **is this the right way to solve the problem?**

**Comment by miguelgrinberg** [comment]
The most common cause for a worker timeout is that you are using blocking code in your application. The solution is to remove or fix the blocking code, not to increase the timeout.
